spdx/parsers/tagvaluebuilders.py

```diff
@@ -698,36 +701,35 @@ def set_pkg_down_location(self, doc, location):
         Raise OrderError if no package previously defined.
         """
         self.assert_package_exists()
-        if not self.package_down_location_set:
-            self.package_down_location_set = True
-            doc.packages[-1].download_location = location
-            return True
-        else:
+        if self.package_down_location_set:
             raise CardinalityError("Package::DownloadLocation")
 
+        self.package_down_location_set = True
+        doc.packages[-1].download_location = location
+        return True
+
     def set_pkg_files_analyzed(self, doc, files_analyzed):
         """
         Set the package files analyzed, if not already set.
         Raise SPDXValueError if malformed value, CardinalityError if
         already defined.
         """
         self.assert_package_exists()
-        if not self.package_files_analyzed_set:
-            if files_analyzed is not None:
-                if validations.validate_pkg_files_analyzed(files_analyzed):
-                    self.package_files_analyzed_set = True
-                    if isinstance(files_analyzed, str):
-                        files_analyzed = files_analyzed.lower() == "true"
-                    doc.packages[-1].files_analyzed = files_analyzed
-                    # convert to boolean;
-                    # validate_pkg_files_analyzed already checked if
-                    # files_analyzed is in ['True', 'true', 'False', 'false']
-                    return True
-                else:
-                    raise SPDXValueError("Package::FilesAnalyzed")
-        else:
+        if self.package_files_analyzed_set:
             raise CardinalityError("Package::FilesAnalyzed")
 
+        if files_analyzed is None or not validations.validate_pkg_files_analyzed(files_analyzed):
```

Comment:
I think the first case is wrong: Before, if `files_analyzed is None`, nothing would happen, right?

Reply:
correct, I fixed it with another early `return True` there